server/item/crossbow.go

```diff
@@ -0,0 +1,166 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"time"
+	_ "unsafe"
+)
+
+// Crossbow is a ranged weapon similar to a bow that uses arrows or fireworks as ammunition.
+type Crossbow struct {
+	// Item is the item the crossbow is charged with.
+	Item Stack
+}
+
+// Charge starts the charging process and checks if the charge duration meets the required duration.
+func (c Crossbow) Charge(releaser Releaser, tx *world.Tx, ctx *UseContext, duration time.Duration) bool {
+	if !c.Item.Empty() {
+		return false
+	}
+
+	creative := releaser.GameMode().CreativeInventory()
+	held, left := releaser.HeldItems()
+
+	chargeDuration := time.Duration(1.25 * float64(time.Second))
+	for _, enchant := range held.Enchantments() {
+		if q, ok := enchant.Type().(interface{ ChargeDuration(int) time.Duration }); ok {
+			chargeDuration = min(chargeDuration, q.ChargeDuration(enchant.Level()))
+		}
+	}
+
+	if duration < chargeDuration {
+		return false
+	}
+
+	var projectileItem Stack
+	if !left.Empty() {
+		_, isFirework := left.Item().(Firework)
+		_, isArrow := left.Item().(Arrow)
+		if isFirework || isArrow {
+			projectileItem = left
+		}
+	}
+
+	if projectileItem.Empty() {
+		var ok bool
+		projectileItem, ok = ctx.FirstFunc(func(stack Stack) bool {
+			_, isArrow := stack.Item().(Arrow)
+			return isArrow
+		})
+
+		if !ok && !creative {
+			return false
+		}
+
+		if projectileItem.Empty() {
+			projectileItem = NewStack(Arrow{}, 1)
+		}
+	}
+
+	c.Item = projectileItem.Grow(-projectileItem.Count() + 1)
+	if !creative {
+		ctx.Consume(c.Item)
+	}
+
+	crossbow := duplicateStack(held, c)
+	releaser.SetHeldItems(crossbow, left)
+	return true
+}
+
+// ReleaseCharge checks if the item is fully charged and, if so, releases it.
+func (c Crossbow) ReleaseCharge(releaser Releaser, tx *world.Tx, ctx *UseContext) bool {
+	if c.Item.Empty() {
+		return false
+	}
+
+	creative := releaser.GameMode().CreativeInventory()
+	rot := releaser.Rotation().Neg()
+	dirVec := releaser.Rotation().Vec3().Normalize()
+
+	if firework, isFirework := c.Item.Item().(Firework); isFirework {
+		createFirework := tx.World().EntityRegistry().Config().Firework
```

Comment:
`SidewaysVelocityMultiplier` & `UpwardsAcceleration` in the firework behavior config should be 0 so the firework flies in a straight path